index.js

```diff
@@ -74,7 +78,7 @@ ${error}`;
         const hex = hashSum.digest('hex');
 
         const artifactClient = new DefaultArtifactClient();
-        const artifactName = hex;
+        const artifactName = isEmptyInput(artifact_name) ? hex : artifact_name + matrix_key;
```

Comment:
I hesitated here to rename the `artifactName` const to something else to differentiate them. However since I noticed inputs are in snake case, I thought maybe that's your convention to differentiate inputs form the rest and that's enough